pallets/credit/src/lib.rs

```diff
@@ -846,6 +848,24 @@ pub mod pallet {
                 }
             }
         }
+
+        fn init_delegator_history(account_id: &T::AccountId, era: u32) -> Weight {
```

Comment:
`init_delegator_history` is only used by `delegate`, so how about return a `Result` not just `Weight`

Reply:
yes,may be returning a bool value is also appropriate